### ISSUE TITLE
Parsing Produces Null If String Is Only One Character

### ISSUE THREAD

**Issue by adamsmd** [ursjoss/KRis#138]
## Example of the Problem

In the following, `startPage` in `ris1` is `null` when it should be "2".  The conditions for this bug to happen seem to require that the lines fed to `toRisRecords()` (1) have no line terminator and (2) contain only one character in their value (e.g., in the sense that "2" has one character but "28" has two).

Example code:
```kotlin
val text1 = listOf(
  "TY  - JOUR",
  "SP  - 2",
  "EP  - 28",
  "ER  - ")
val ris1 = text1.toRisRecords()
val text2 = listOf(
  "TY  - JOUR\n",
  "SP  - 2\n",
  "EP  - 28\n",
  "ER  - \n")
val ris2 = text2.toRisRecords()
println("ris1.startPage: <${ris1.first().startPage}>")
println("ris1.endPage: <${ris1.first().endPage}>")
println("ris2.startPage: <${ris2.first().startPage}>")
println("ris2.endPage: <${ris2.first().endPage}>")
```

Output from that example code:
```
ris1.startPage: <null>
ris1.endPage: <28>
ris2.startPage: <2>
ris2.endPage: <28>
```

## Comments

This is not a major blocker for my work because, after figuring out what is going on, it is easy enough to make my line splitting leave the "\n" at the end of the strings.  However, it was a surprising behavior and is not compatible with things like [kotlin.text.lines()][https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/lines.html], which do not leave the "\n" at the end, so I would recommend KRis allow lines without a line terminator.

Also, I think this causes a bug in `KRisIO.process()`, which uses `readLines()`, which (as I understand it) omits line endings.  My testing of that hasn't been as complete, so I might be wrong about that, though.

**Comment by ursjoss** [comment]
Hi @adamsmd , thanks again for your well-composed ticket. I'll have a look into it soon.

**Comment by ursjoss** [comment]
@adamsmd I have released [KRis-0.4.3](https://central.sonatype.com/artifact/ch.difty.kris/kris-core/0.4.3/versions). Please check if it resolves your issue...

**Comment by adamsmd** [comment]
Yes, that resolves the issue.  Thank you.
